src/lib/address_resolve/AddressResolve.cpp

```diff
@@ -0,0 +1,24 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <lib/address_resolve/AddressResolve.h>
+
+namespace chip {
+namespace AddressResolve {
+
+} // namespace AddressResolve
+} // namespace chip
```

Comment:
Why do we need this empty file?

Reply:
I will try to add something to it. Generally I believe ninja does not like static_library without cpp files (in case the implementation is 'custom'). 

Will add a comment.

Reply:
"I believe ninja does not like static_library without cpp files"

The impl build target chosen should be the one with the cpp file. The interfaces can be part of a different target that is only a dependency chain of headers, and does not yield a library. 

I feel like there is a way around this, if we ask someone more familiar with gn